components/button/Button.js

```diff
@@ -120,11 +120,22 @@ const factory = (ripple, FontIcon) => {
         'data-react-toolbox': 'button',
       };
 
-      return React.createElement(element, props,
+      const renderElementWrapper = element === 'button' && others.onMouseEnter && this.props.disabled;
+      const componentToRender = React.createElement(element, props,
         icon ? <FontIcon className={theme.icon} value={icon} /> : null,
         label,
         children,
       );
+
+      if (renderElementWrapper) {
```

Comment:
I'd do something like this, extracting the mouse events to a variable to reuse it and that allows you to return a ternary, but that's just code style.

```js
const mouseEvents = {
  onMouseUp: this.handleMouseUp,
  onMouseLeave: this.handleMouseLeave,
};

const props = {
  ...others,
  ...mouseEvents,
  href,
  ref: (node) => { this.buttonNode = node; },
  className: classes,
  disabled: this.props.disabled,
  type: !href ? type : null,
  'data-react-toolbox': 'button',
};

const buttonElement = React.createElement(element, props,
  icon ? <FontIcon className={theme.icon} value={icon} /> : null,
  label,
  children,
);

return others.onMouseEnter && this.props.disabled
  ? <span {...mouseEvents}>{buttonElement}</span>
  : buttonElement;
```

Reply:
Updated the PR with your code suggestions. Thank you for take a look on this.